test/minitest/test_minitest_mock.rb

```diff
@@ -495,4 +495,18 @@ def test_mock_with_yield
     @tc.assert_equal true, rs
   end
 
+  def test_blow_up_on_multiple_stub
+    h = {}
+    e = assert_raises StubError do
+      h.stub(:empty?, false) do
+        h.stub(:empty?, true) do
```

Comment:
Do you have a concrete example of that situation? I feel you should be able to work around it by narrowing the scope of what you're testing in each test, through dependency injection and mocking, or some other testing/design practice that will make the use of nested stubs like that, pointless.

In any case, I'm almost certain you can find a refactor that will make it easier to test that situation.